backpack/utils/linear.py

```diff
@@ -1,17 +1,60 @@
-from torch import einsum
+"""Contains utility functions to extract the GGN diagonal for linear layers."""
+from torch import Tensor, einsum
+from torch.nn import Linear
 
 
-def extract_weight_diagonal(module, backproped, sum_batch=True):
-    if sum_batch:
-        equation = "vno,ni->oi"
+def extract_weight_diagonal(
+    module: Linear, S: Tensor, sum_batch: bool = True
```

Comment:
`S` is a bad name because it contains capital letters and does not explain anything. I guess it is extension specific and therefore fine?

Reply:
I renamed it from `sqrt_ggn` to `S`, which is closer to the paper notation. One could also take the opposite view and argue that for internals it may be helpful to stay close to the paper notation.

Marking as resolved, but feel free to reopen if you think it should be addressed.